main_comment.go

```diff
@@ -147,31 +147,40 @@ func processGitHubComment(
 //	}
 func parsePrOptions(commentBody string) (map[string]string, error) {
 	prRepos := make(map[string]string)
-	var err error = nil
-
+	var err error
 	words := strings.Fields(commentBody)
 	tokensCount := len(words)
-	for i, w := range words {
-		if w == "--pr" && i < (tokensCount-1) {
-			v := words[i+1]
-			o := strings.Index(v, "/")
-			if o > 0 {
-				skip := 0
-				if v[o:][0] == '/' && len(v[o:]) > 1 {
-					skip = 1
-				}
-				revision := v[o+skip:]
-				if _, err := strconv.Atoi(revision); err == nil {
-					revision = "pull/" + revision + "/head"
+	for id, word := range words {
+		if word == "--pr" && id < (tokensCount-1) {
+			userInput := strings.TrimSpace(words[id+1])
+			userInputParts := strings.Split(userInput, "/")
+
+			if len(userInput) > 0 {
+				var revision string
+				switch len(userInputParts) {
+				case 2: // we can have both deviceauth/1 and mender/3.1.x syntax
+					// repo/<pr_number> syntax
+					if _, err := strconv.Atoi(userInputParts[1]); err == nil {
+						revision = "pull/" + userInputParts[1] + "/head"
+					} else {
+						// feature branch
+						prPartsAndRandString := strings.Split(userInputParts[1], " ")
```

Comment:
I don't understand why splitting on spaces here is necessary. There should not be any, or? Is it just a leftover from the previous implementation? Strictly speaking it does no harm, it's just unnecessary, and you should be able to just write this:

```golang
revision = userInputParts[1]
```

Reply:
Good catch, thank you!